### PR TITLE
hbacrule: Allow clearing members with empty lists.

### LINKED ISSUE #751: hbacrule: can't force a member list to be empty.
If I have an `hbacrule` with some `hbacsvc`, and I want to
ensure that `hbacsvc` is empty, I have to create a task with
all the `hbacsvc` and `state: absent`.

I'd expect that using something like the task below, the
`hbacsvc` member would be cleared:
```yaml
- ipahbacrule:
    ipaadmin_password: SomeADMINpassword
    name: hbacrule01
    hbacsvc: []
```

The same behavior is verified in the other members.

### PR DESCRIPTION
**Pull request by rjeffman**
If a hbacrule member has any value, the only way to clear it is by
creating a task with 'state: absent' and 'action: member' and provide
a list with all the values for that member.

This patch allows the use of '<member>: []' with 'action: hbacrule'
to clear a hbacrule member.

Fix #751

### CODE REVIEW

File: tests/hbacrule/test_hbacrule.yml [hunk 1]
```diff
@@ -662,6 +662,129 @@
...
+      ipaapi_context: "{{ ipa_context | default(omit) }}"
+      name: hbacrule01
+      host:
+      - "{{ 'testhost03.' + ipaserver_domain }}"
+      - "{{ 'testhost04.' + ipaserver_domain }}"
+      hostgroup: testhostgroup03,testhostgroup04
+      user: testuser03,testuser04
+      group: testgroup03,testgroup04
+      hbacsvc: testhbacsvc03,testhbacsvc04
+      hbacsvcgroup: testhbacsvcgroup03,testhbacsvcgroup04
```
Comment (t-woerner) [inline tests/hbacrule/test_hbacrule.yml:678]: It might be good to add a test in the end to make sure that any of these entries are not there anymore after using the empty lists.

Comment (rjeffman) [reply tests/hbacrule/test_hbacrule.yml]: I added two tests with `check_mode: yes` to verify if the attributes are empty.

I also refactored the playbook test so that the tests for empty members are in its own test playbook, reducing the size of the base test (which is still too big, IMO).

**Review by t-woerner — APPROVED** [review]
LGTM
